libraries/Ethernet/src/utility/ethernetif.c

```diff
@@ -128,9 +128,11 @@ void HAL_ETH_MspInit(ETH_HandleTypeDef *heth)
     }
   }
 
+#ifdef ETH_INPUT_USE_IT

```

Comment:
Yes it works with and without. But it is slower than the polling mode. And LwIP stack documentation recommends to use the polling mode.
It gives to the user the possibility to use it.